### PR TITLE
Baseline Dmitry #580

### PR DESCRIPTION
**Pull request by Oliver2049**
- Added required imports
- Added the appropriate docstring to the functions and interface Changed the state variables to camelCase
- Added inline comments to all react hooks and to all component constant 
- Removed the Userguide(title,description_userguide) in Stack as it is no longer required 
- Added the steps for user guide in component description 
- Add required component state variables
- Added sourceLink and updated it with the appropriate link Added empty string in tutorial
- Fixed grammar issues
- Added the render component for user guide which was not there in the code 
- Run prettier

### CODE REVIEW

**Comment by michaeljpigott** [comment]
Hi @Oliver2049 

Can you please clarify why you are updating FTP connect in a pull request about baselining Dmitry? 

Kind regards 

Michael

**Comment by Oliver2049** [comment]
> Hi @Oliver2049
> 
> Can you please clarify why you are updating FTP connect in a pull request about baselining Dmitry?
> 
> Kind regards
> 
> Michael

Thanks for your feedback, Michael. I managed to fix and update the tool. You can check again.
Regards, Oliver.

File: src/components/Dmitry/Dmitry.tsx [hunk 1]
```diff
@@ -1,45 +1,79 @@
...
+    // Component Constants.
+    const title = "Deepmagic Information Gathering Tool"; // Title of the component.
+    const description =
+        "DMitry is a UNIX/(GNU) Linux command line application written in C. DMitry will scan to try and find " +
+        "possible subdomains, email addresses and uptime information, and further perform tcp port scans, and " +
+        "whois lookups."; // Description of the component.
+    const steps =
+        "Step 1: Enter a valid domain or IP address to be scanned.\n" +
+        "       Eg: 192.168.0.1\n\n" +
+        "Step 2: Enter a delay between requests. Default is 2 (milliseconds). Can be left blank.\n" +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:53]: Which field are you meant to put the delay between requests? It is not clear.

File: src/components/Dmitry/Dmitry.tsx [hunk 2]
```diff
@@ -1,45 +1,79 @@
...
+        "       Eg: 2\n\n" +
+        "Step 3: Click Start Scanning to commence Dmitry's operation.\n\n" +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:55]: There should be at least one step that refers to DMitry's advanced options.

File: src/components/Dmitry/Dmitry.tsx [hunk 3]
```diff
@@ -1,45 +1,79 @@
...
+        "Step 4: View the Output block below to view the results of the tool's execution.\n";
+    const sourceLink = "https://www.kali.org/tools/dmitry/\n"; // Link to the source code (or Kali Tools).
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:57]: This link is broken. You do not need the \n. 
```suggestion
    const sourceLink = "https://www.kali.org/tools/dmitry/"; // Link to the source code (or Kali Tools).
```

File: src/components/Dmitry/Dmitry.tsx [hunk 4]
```diff
@@ -156,97 +226,110 @@ const dmitry = () => {
...
+                                checked={checkedDomainname}
+                                onChange={(e) => setCheckedDomainname(e.currentTarget.checked)}
+                            />
+                            <Checkbox
+                                label={"Retrieve Netcraft.com information on a host."}
+                                checked={checkedNetcraft}
+                                onChange={(e) => setCheckedNetcraft(e.currentTarget.checked)}
+                            />
+                            <Checkbox
+                                label={"Search for a possible subdomains."}
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:289]: Typo. Should be "Search for possible subdomains."

File: src/components/Dmitry/Dmitry.tsx [hunk 5]
```diff
@@ -156,97 +226,110 @@ const dmitry = () => {
...
+                                        type="number"
+                                        value={delay}
+                                        onChange={(e) => setDelay(parseInt(e.target.value))}
+                                    />
+                                </div>
+                            )}
+                        </>
+                    )}
+
+                    <Button type={"submit"}>Start Scanning</Button>
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:327]: Should have the tool's title. 
```suggestion
                    <Button type={"submit"}>Start { Title }</Button>
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @Oliver2049 

Good work. I recommend updating the steps to they are a bit more descriptive and include a reference to the advanced options. 

Kind regards

Michael

**Comment by michaeljpigott** [comment]
Hi @Oliver2049 I am still waiting for you to implement the requested changes. 

Kind regards 

Michael

File: src/components/Dmitry/Dmitry.tsx [hunk 6]
```diff
@@ -1,45 +1,79 @@
...
+    const description =
+        "Dmitry is a UNIX/(GNU) Linux command line application written in C. Dmitry will scan to try and find " +
+        "possible subdomains, email addresses and uptime information, and further perform tcp port scans, and " +
+        "whois lookups."; // Description of the component.
+    const steps =
+        "Step 1: Enter a valid domain or IP address to be scanned.\n" +
+        "       Eg: 192.168.0.1\n\n" +
+        "Step 3: Click Start Scanning to commence Dmitry's operation.\n\n" +
+        "Step 4: View the Output block below to view the results of the tool's execution.\n" +
+        "Note: For more advanced options, enable the Advanced Mode switch to access additional features.\n" +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:55]: Extra line break.
```suggestion
        "Note: For more advanced options, enable the Advanced Mode switch to access additional features.\n\n" +
```

File: src/components/Dmitry/Dmitry.tsx [hunk 7]
```diff
@@ -1,45 +1,79 @@
...
+    // Component Constants.
+    const title = "Deepmagic Information Gathering Tool"; // Title of the component.
+    const description =
+        "Dmitry is a UNIX/(GNU) Linux command line application written in C. Dmitry will scan to try and find " +
+        "possible subdomains, email addresses and uptime information, and further perform tcp port scans, and " +
+        "whois lookups."; // Description of the component.
+    const steps =
+        "Step 1: Enter a valid domain or IP address to be scanned.\n" +
+        "       Eg: 192.168.0.1\n\n" +
+        "Step 3: Click Start Scanning to commence Dmitry's operation.\n\n" +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:53]: Need to change the step numbering. 
```suggestion
        "Step 2: Click Start Scanning to commence Dmitry's operation.\n\n" +
```

File: src/components/Dmitry/Dmitry.tsx [hunk 8]
```diff
@@ -1,45 +1,79 @@
...
+        "Step 4: View the Output block below to view the results of the tool's execution.\n" +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:54]: Need to change step numbering. 
```suggestion
        "Step 3: View the Output block below to view the results of the tool's execution.\n" +
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @Oliver2049 

This is looking good. Thank you for making those changes and I have requested a few more minor changes. 

Kind regards

Michael

File: src/components/Dmitry/Dmitry.tsx [hunk 9]
```diff
@@ -1,45 +1,81 @@
...
+    // Component Constants.
+    const title = "Deepmagic Information Gathering Tool"; // Title of the component.
+    const description =
+        "Dmitry is a UNIX/(GNU) Linux command line application written in C. Dmitry will scan to try and find " +
+        "possible subdomains, email addresses and uptime information, and further perform tcp port scans, and " +
+        "whois lookups."; // Description of the component.
+    const steps =
+        "Step 1: Enter a valid domain or IP address to be scanned.\n" +
+        "       Eg: 192.168.0.1\n\n" +
+        "Step 3: Click Start " +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:53]: You still need to change the numbering. 
```suggestion
        "Step 2: Click Start " +
```

File: src/components/Dmitry/Dmitry.tsx [hunk 10]
```diff
@@ -1,45 +1,81 @@
...
+        title +
+        " to commence Dmitry's operation.\n\n" +
+        "Step 4: View the Output block below to view the results of the tool's execution.\n\n" +
```
Comment (michaeljpigott) [inline src/components/Dmitry/Dmitry.tsx:56]: Numbering. 
```suggestion
        "Step 3: View the Output block below to view the results of the tool's execution.\n\n" +
```

**Review by michaeljpigott — CHANGES_REQUESTED** [review]
Hi @Oliver2049 

I think you forgot to update the numbering of the steps - it goes from Step 1 to Step 3. 

Kind regards

Michael

File: src/components/Dmitry/Dmitry.tsx (hunk 10 above)
Comment (Oliver2049) [reply src/components/Dmitry/Dmitry.tsx:56]: Hi @michaeljpigott
Well that's quite strange as i did update them, but no worries, i will fix this.
Regards, 
Oliver

**Review by michaeljpigott — APPROVED** [review]
Hi @Oliver2049 

Thank you for making those changes. I approve this pull request. 

Kind regards

Michael

**Pull request merged**